Build fails on MacOS with java and hdfs support

<!--
IMPORTANT: Do NOT use GitHub to post any questions or support requests!
           They will be closed immediately and ignored.

Questions should go to the gdal-dev mailing list at
https://lists.osgeo.org/mailman/listinfo/gdal-dev
or other support forums. GitHub issues are for bug
reports and suggestions for new features.

If you think there is an issue with coordinate order in GDAL 3.0, then it is
likely an intended behaviour. See https://github.com/OSGeo/gdal/issues/1974
for more explanations.

The GDAL project is made of contributions from various individuals and
organizations, each with their own focus. The issue you are facing is
not necessarily in the priority list of those contributors and consequently
there is no guarantee that it will be addressed in a timely manner.
If this bug report or feature request is high-priority for you, and you
cannot address it yourself, we suggest engaging a GDAL developer or support
organisation and financially sponsoring a fix.

-->

## Expected behavior and actual behavior.

Build success.
Supporting `libjvm.dylib` to build.

## Steps to reproduce the problem.

```bash
./configure --prefix=/usr/local/Cellar/gdal/2.4.0 --with-java=$JAVA_HOME --with-hdfs=$HADOOP_HOME
```
output:
```bash
checking GDAL version information from gdal_version.h.in: 2.4.0
checking for Mac OS X Framework build... disabled
checking for perl bindings... disabled
checking for python bindings... disabled
checking whether we should include Java support... yes
configure: error: Unable to find libjvm.so.
```
![image](https://user-images.githubusercontent.com/6694539/76510106-088ed980-648c-11ea-9741-b4b7dc8b47dd.png)
In above image, i only have libjvm.dylib instead of libjvm.so. And i check the `configure.ac` file, it doesn't consider of `libjvm.dylib`. [related link in configure.ac](https://github.com/OSGeo/gdal/blob/9d4d488a492510fdf3b20d8b74aba06ebeb5ba76/gdal/configure.ac#L5387)
## Operating system

Mac OSX 10.15.3

## GDAL version and provenance

2.4.0

--- a/gdal/configure
+++ b/gdal/configure
@@ -941,7 +941,6 @@ infodir
 docdir
 oldincludedir
 includedir
-runstatedir
 localstatedir
 sharedstatedir
 sysconfdir
@@ -1302,7 +1301,6 @@ datadir='${datarootdir}'
 sysconfdir='${prefix}/etc'
 sharedstatedir='${prefix}/com'
 localstatedir='${prefix}/var'
-runstatedir='${localstatedir}/run'
 includedir='${prefix}/include'
 oldincludedir='/usr/include'
 docdir='${datarootdir}/doc/${PACKAGE}'
@@ -1555,15 +1553,6 @@ do
   | -silent | --silent | --silen | --sile | --sil)
     silent=yes ;;
 
-  -runstatedir | --runstatedir | --runstatedi | --runstated \
-  | --runstate | --runstat | --runsta | --runst | --runs \
-  | --run | --ru | --r)
-    ac_prev=runstatedir ;;
-  -runstatedir=* | --runstatedir=* | --runstatedi=* | --runstated=* \
-  | --runstate=* | --runstat=* | --runsta=* | --runst=* | --runs=* \
-  | --run=* | --ru=* | --r=*)
-    runstatedir=$ac_optarg ;;
-
   -sbindir | --sbindir | --sbindi | --sbind | --sbin | --sbi | --sb)
     ac_prev=sbindir ;;
   -sbindir=* | --sbindir=* | --sbindi=* | --sbind=* | --sbin=* \
@@ -1701,7 +1690,7 @@ fi
 for ac_var in	exec_prefix prefix bindir sbindir libexecdir datarootdir \
 		datadir sysconfdir sharedstatedir localstatedir includedir \
 		oldincludedir docdir infodir htmldir dvidir pdfdir psdir \
-		libdir localedir mandir runstatedir
+		libdir localedir mandir
 do
   eval ac_val=\$$ac_var
   # Remove trailing slashes.
@@ -1854,7 +1843,6 @@ Fine tuning of the installation directories:
   --sysconfdir=DIR        read-only single-machine data [PREFIX/etc]
   --sharedstatedir=DIR    modifiable architecture-independent data [PREFIX/com]
   --localstatedir=DIR     modifiable single-machine data [PREFIX/var]
-  --runstatedir=DIR       modifiable per-process data [LOCALSTATEDIR/run]
   --libdir=DIR            object code libraries [EPREFIX/lib]
   --includedir=DIR        C header files [PREFIX/include]
   --oldincludedir=DIR     C header files for non-gcc [/usr/include]
@@ -41087,7 +41075,7 @@ if test -z "$with_hdfs" -o "$with_hdfs" = "no" ; then
 $as_echo "HDFS support not requested." >&6; }
 else
      if test "$JAVA_HOME"; then
-          JNI_LIB="-L$(find $JAVA_HOME | grep libjvm.so | head -1 | sed 's/libjvm.so$//') -ljvm"
+          JNI_LIB="-L$(find $JAVA_HOME | grep libjvm | head -1 | sed "s/[^/]*$//") -ljvm"
      else
           as_fn_error $? "HDFS support requires java support (--with-java). $JAVA_HOME" "$LINENO" 5
      fi
@@ -41169,7 +41157,7 @@ $as_echo "HDFS support enabled." >&6; }
                as_fn_error $? "HDFS support not enabled." "$LINENO" 5
           fi
      else
-          as_fn_error $? "Unable to find libjvm.so." "$LINENO" 5
+          as_fn_error $? "Unable to find libjvm." "$LINENO" 5
      fi
 fi
 

--- a/gdal/configure.ac
+++ b/gdal/configure.ac
@@ -5384,7 +5384,7 @@ if test -z "$with_hdfs" -o "$with_hdfs" = "no" ; then
      AC_MSG_RESULT([HDFS support not requested.])
 else
      if test "$JAVA_HOME"; then
-          JNI_LIB="-L$(find $JAVA_HOME | grep libjvm.so | head -1 | sed 's/libjvm.so$//') -ljvm"
+          JNI_LIB="-L$(find $JAVA_HOME | grep libjvm | head -1 | sed "s/[[^\/]]*$//") -ljvm"
      else
           AC_MSG_ERROR([HDFS support requires java support (--with-java). $JAVA_HOME])
      fi
@@ -5401,7 +5401,7 @@ else
                AC_MSG_ERROR([HDFS support not enabled.])
           fi
      else
-          AC_MSG_ERROR([Unable to find libjvm.so.])
+          AC_MSG_ERROR([Unable to find libjvm.])
      fi
 fi
 

--- a/gdal/configure.ac
+++ b/gdal/configure.ac
@@ -5384,7 +5384,7 @@ if test -z "$with_hdfs" -o "$with_hdfs" = "no" ; then
      AC_MSG_RESULT([HDFS support not requested.])
 else
      if test "$JAVA_HOME"; then
-          JNI_LIB="-L$(find $JAVA_HOME | grep libjvm | head -1 | sed "s/[[^\/]]*$//") -ljvm"
+          JNI_LIB="-L$(find $JAVA_HOME | grep libjvm | head -1 | sed "s/[[^/]]*$//") -ljvm"
      else
           AC_MSG_ERROR([HDFS support requires java support (--with-java). $JAVA_HOME])
      fi

